### PR TITLE
refactor: sloads and increments in assembly for gas optimization. #179

### LINKED ISSUE #179: refactor: sloads and increments in assembly for gas optimization.
### Describe the Refactor Request

sloads and increments can be written in assembly for gas optimization.

### PR DESCRIPTION
**Pull request by blackbeard002**
closes #179

### CODE REVIEW

**Comment by blackbeard002** [comment]
Hey @0xneves ready for review

File: contracts/Swaplace.sol [hunk 1]
```diff
@@ -26,14 +26,16 @@ contract Swaplace is SwapFactory, ISwaplace, IERC165 {
   function createSwap(Swap calldata swap) public returns (uint256) {
     if (swap.owner != msg.sender) revert InvalidAddress(msg.sender);
 
+    uint256 swapId;
+
     unchecked {
       assembly {
         sstore(_totalSwaps.slot, add(sload(_totalSwaps.slot), 1))
+
+        swapId := sload(_totalSwaps.slot)
```
Comment (0xneves) [inline contracts/Swaplace.sol:35]: This won't change gas usage, we can use the previous way for better legibility

File: contracts/Swaplace.sol [hunk 2]
```diff
@@ -65,7 +67,13 @@ contract Swaplace is SwapFactory, ISwaplace, IERC165 {
         assets[i].amountOrId
       );
       unchecked {
-        i++;
+        assembly {
+          i := mload(0x40)
+
```
Comment (0xneves) [inline contracts/Swaplace.sol:72]: Remove line 72

File: contracts/Swaplace.sol [hunk 3]
```diff
@@ -65,7 +67,13 @@ contract Swaplace is SwapFactory, ISwaplace, IERC165 {
...
+          i := add(i, 1)
+
```
Comment (0xneves) [inline contracts/Swaplace.sol:74]: Remove line 74

File: contracts/Swaplace.sol [hunk 4]
```diff
@@ -78,7 +86,13 @@ contract Swaplace is SwapFactory, ISwaplace, IERC165 {
         assets[i].amountOrId
       );
       unchecked {
-        i++;
+        assembly {
+          i := mload(0x40)
+
```
Comment (0xneves) [inline contracts/Swaplace.sol:91]: Remove line 91

File: contracts/Swaplace.sol [hunk 5]
```diff
@@ -78,7 +86,13 @@ contract Swaplace is SwapFactory, ISwaplace, IERC165 {
...
+          i := add(i, 1)
+
```
Comment (0xneves) [inline contracts/Swaplace.sol:93]: Remove line 93

File: contracts/Swaplace.sol [hunk 6]
```diff
@@ -126,6 +140,14 @@ contract Swaplace is SwapFactory, ISwaplace, IERC165 {
    * @dev Getter function for _totalSwaps.
    */
   function totalSwaps() public view returns (uint256) {
-    return _totalSwaps;
+    uint256 swapId;
+
```
Comment (0xneves) [inline contracts/Swaplace.sol:144]: totalSwaps is never called on-chain, therefore there is no gas optm., we can keep the original way to preserve legibility

**Review by 0xneves — CHANGES_REQUESTED** [review]
Requesting the following changes:

**Comment by blackbeard002** [comment]
Hey @0xneves, done as requested. Ready for review.

**Review by 0xneves — APPROVED** [review]

**Pull request merged**